robolectric/src/main/java/org/robolectric/android/internal/LocalUiController.java

```diff
@@ -49,9 +50,10 @@ public boolean injectMotionEvent(MotionEvent event) throws InjectEventSecurityEx
   public boolean injectKeyEvent(KeyEvent event) throws InjectEventSecurityException {
     checkNotNull(event);
     checkState(Looper.myLooper() == Looper.getMainLooper(), "Expecting to be on main thread!");
-
     loopMainThreadUntilIdle();
-    getViewRoot().dispatchKeyEvent(event);
+
+    // FLAG_NOT_FOCUSABLE: "this window won't ever get key input focus"
+    getTopMostViewRootExcluding(LayoutParams.FLAG_NOT_FOCUSABLE).dispatchKeyEvent(event);
```

Comment:
Hi @wesalvaro, could add related tests for your fix occasion? It's better if you add those tests to `ctesque` module that run with `Robolectric` as virtual emulator, and real emulator.

Reply:
Thanks for pointing this out, the tests are contained in the EspressoWithWindowLayersTest file in this PR.

Reply:
What about running those tests on CI with emulator too?

Reply:
Does that involve making integration_tests/androidx_test follow the sharedTest pattern? I think all of integration_tests/androidx_test  should work as-is on emulators.

Reply:
Yes. We can use sharedTest pattern to restructure androidx_test. Maybe it can be left as a new problem. I will fire an issue for it today.